### PR TITLE
New article - Introduction to character encoding

### PR DESCRIPTION
**Pull request by tdykstra**
Related to #15845 

[Internal review URL](https://review.docs.microsoft.com/en-us/dotnet/standard/base-types/character-encoding-introduction?branch=pr-en-us-17448)

### CODE REVIEW

**Comment by GrabYourPitchforks** [comment]
Came across this when reviewing https://github.com/dotnet/dotnet-api-docs/pull/3995 - thank you so much for adapting the draft and making it more approachable! :)

The only real feedback I have is that we should be careful with how we use the word _character_. We're a bit loose with it, sometimes using it to mean to an assigned code point, other times using it to mean a display character ("grapheme cluster"). If we had a chance to clean this up before publishing I think this would help.

**Review by BillWagner — APPROVED** [review]
This LGTM.

You can :shipit: once you decide how to address the comments from @GrabYourPitchforks

**Review by GrabYourPitchforks — APPROVED** [review]

File: docs/standard/base-types/character-encoding/csharp/ConvertToUpper.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+﻿using System;
```
Comment (Thraka) [inline docs/standard/base-types/character-encoding/csharp/ConvertToUpper.cs]: These files should be hosted in a `snippets` folder, just like `includes` or `images`. Also, should the name of the folder be `character-encoding-introduction` ? So the path is:

```
docs/standard/base-types/snippets/character-encoding-introduction/csharp/*
```

**Review by Thraka — DISMISSED** [review]
Looks great!

The snippets should be relocated. Other things are just minor changes or suggestions.

File: docs/standard/base-types/character-encoding-introduction.md [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+
+The built-in `Encoding` classes can also be configured to throw an exception rather than perform character substitution when ill-formed sequences are seen. This approach is often used in security-sensitive applications where character substitution might not be acceptable.
+
+```csharp
+byte[] utf8Bytes = ReadFromNetwork();
+UTF8Encoding encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
```
[the remark this answers is not in the document]
Comment (tdykstra) [reply docs/standard/base-types/character-encoding-introduction.md]: Back at the start of this section it gives an example of ill-formed UTF8 encoding.  It's not the same thing as endianness, which is about communication between different machine architectures. I'll add a sentence noting that.

**Pull request merged**